### PR TITLE
UPDATE elasticache module

### PR DESCRIPTION
**Pull request by phoolish**
* Add create_parameter group var
* Add external security group var
* Add elasticache test
* Make parameter group and security group names unique
* Add create before destroy lifecycle rules

### CODE REVIEW

File: aws/elasticache/main.tf [hunk 1]
```diff
@@ -44,38 +46,51 @@ resource "aws_elasticache_replication_group" "mod" {
...
+    create_before_destroy = true
+  }
 }
 
 resource "aws_security_group" "sg_for_access_by_sgs" {
-  name = "${var.env}-${var.engine}"
-  description = "${var.env} to ${var.engine}"
+  count = "${var.provide_sg_for_access ? 1 : 0}"
+  name = "${var.name}_${var.env}-${var.engine}"
+  description = "${var.name}_$${var.env} to ${var.engine}"
```
Comment (pturley) [inline aws/elasticache/main.tf]: extra $ here. Found that because of the planshots :)

Comment (phoolish) [reply aws/elasticache/main.tf]: Ha. Or you could have said thanks for making planshots more usable :)

**Comment by phoolish** [comment]
@pturley second look?

**Comment by pturley** [comment]
Needs a rebase and fixup of the planshots

File: aws/elasticache/outputs.tf [hunk 2]
```diff
@@ -1,5 +1,5 @@
 output "sg_for_access_by_sgs_id" {
-  value = "${aws_security_group.sg_for_access_by_sgs.id}"
+  value = "${element(concat(aws_security_group.sg_for_access_by_sgs.*.id, list("")), 0)}"
```
Comment (pturley) [inline aws/elasticache/outputs.tf:2]: I dont understand this line

**Review by pturley — APPROVED** [review]

File: aws/elasticache/outputs.tf (hunk 2 above)
Comment (phoolish) [reply aws/elasticache/outputs.tf:2]: This allows for an empty output otherwise terraform complains

**Pull request merged**